cudax/include/cuda/experimental/stf.cuh

```diff
@@ -473,17 +463,10 @@ public:
       payload);
   }
 
+#if defined(CUDASTF_CODE_GENERATION) && defined(__CUDACC__)
```

Comment:
I'm wondering if we should still define the launch method so that we can keep the static assertion, but then we need to define thread_hierarchy_spec as well even if we can't use it, so the benefits of the static_assertion are little compared to the actual complications....

Maybe we can just do
```
#ifdef CODEGEN
template <typename thread_hierarchy_spec_t, typename... Deps>
auto launch(thread_hierarchy_spec_t spec, exec_place e_place, task_dep<Deps>... deps) {
   ...
}
#else
template <Args...>
void launch(Args...) {
    static_assert(...)
}
#endif
```

Reply:
But it won't help compile time and adds some complications for maintenance

Reply:
I've removed the static_assertion unit tests, the value is far below the unnecessary complications